[frontend] Individual question

carriezhengjr opened the pull request:
Allow user to view individual question when view question button is clicked.
Fixes #8

carriezhengjr committed:
View individual question
Allow user to click question view button,
and view details of each question.

carriezhengjr committed:
Format style such as bold headings and font size

carriezhengjr committed:
Styling changes: adjust paddings

carriezhengjr committed:
Merge branch 'master' into individual-question

carriezhengjr committed:
Adjust code spacing issues according to Prettier

carriezhengjr committed:
Edit styling of complexity tag
Align complexity tag style with the one shown in question list

chloeelim commented on frontend/src/pages/questions/ViewQuestion.tsx:
```diff
@@ -0,0 +1,66 @@
+        <Heading as="h2" size="md">
+          Description
+        </Heading>
+        <Text>{question.questionDescription}</Text>
```
Hi @carriezhengjr ,

Looks nice! One small suggestion here so that line breaks in the question description are being displayed properly!

```suggestion
        <Text whiteSpace="pre-line">{question.questionDescription}</Text>
```

| Previous | After |
|----------|------|
|<img width="1427" alt="Screenshot 2023-09-13 at 9 08 32 PM" src="https://github.com/CS3219-AY2324S1/ay2324s1-course-assessment-g17/assets/79991214/407e60cb-7f5b-4de4-bafd-b8a918b8dd2a">|<img width="1416" alt="Screenshot 2023-09-13 at 9 08 16 PM" src="https://github.com/CS3219-AY2324S1/ay2324s1-course-assessment-g17/assets/79991214/1956fa39-e17e-4bd4-b049-c1a132c88741">|

carriezhengjr committed:
Display line breaks properly

Everything appears as a single paragraph without line breaks.

Handle whitespaces and line breaks so that
sequences of whitespace will collapse into a single whitespace, and
text will wrap when necessary and on line breaks.

sltsheryl reviewed (approved):
LGTM
